File: Python/longestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        size = 0
        check = set(nums)

        for num in nums:
            if num - 1 not in check:
                count = 0
                curr = num + 1
                while curr in check:
                    count += 1
                    curr += 1
                size = max(count, size)

        return size + 1

File: Java/longestConsecutiveSequence.java
class Solution {
    public int longestConsecutive(int[] nums) {

        if(nums.length==0)
            return 0;
            
        int size= 0;
        HashSet<Integer> check= new HashSet<>();

        for(int i:nums)
        check.add(i);

        int n= nums.length;
        int i=0;

        while(i<n){
            if(!check.contains(nums[i]-1)){
                int count= 0;
                int curr= nums[i]+1;
                if(check.contains(curr)){
                    while(check.contains(curr)){
                        count+=1;
                        curr+=1;
                    }
                }
                size= Math.max(count, size);
            }
            
            i+=1;
        }

        return size+1;
    }
}